fs/procfs/fs_procfs.c

```diff
@@ -184,27 +184,24 @@ static const struct procfs_entry_s g_procfs_entries[] =
 #if defined(CONFIG_DEBUG_TCBINFO) && !defined(CONFIG_FS_PROCFS_EXCLUDE_TCBINFO)
   { "tcbinfo",       &tcbinfo_operations,         PROCFS_FILE_TYPE   },
 #endif
+  { NULL,            NULL,                        PROCFS_UNKOWN_TYPE },
 };
 
 #ifdef CONFIG_FS_PROCFS_REGISTER
 static const uint8_t g_base_entrycount = sizeof(g_base_entries) /
```

Comment:
why do we need `g_base_entrycount` if array us already NULL terminated and can be just `while`-ed till `pathpattern != NULL`?

Reply:
Adding NULL entry avoid the out of boundary error:
```
procfs/fs_procfs.c: In function 'procfs_readdir':
Error: procfs/fs_procfs.c:878:35: error: array subscript <unknown> is outside array bounds of 'const struct procfs_entry_s[0]' [-Werror=array-bounds]
  878 |       if (strncmp(g_procfs_entries[level1->base.index].pathpattern,
```
Many place verify the index in the range, the code has to recompute every time without `g_base_entrycount`.

Reply:
Now we have a similar error, but at line 740